test/Error/aggregate_error.js

```diff
@@ -0,0 +1,384 @@
+
+            var case4 = {
+                [Symbol.iterator]() {
+                  return "a string";
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case4);
+            }, TypeError);
+
+            var case5 = {
+                [Symbol.iterator]() {
+                    return undefined;
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case5);
+            }, TypeError);
+
+            var case6 = {
+                [Symbol.iterator]() {
+                    return {
+                        get next() {
+                            throw new TestError();
+                        }
+                    }
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case6);
+            }, TestError);
+
+            var case7 = {
+                [Symbol.iterator]() {
+                    return {
+                        get next() {
+                            return {};
+                        }
+                    }
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case7);
+            }, TypeError);
+
+            var case8 = {
+                [Symbol.iterator]() {
+                    return {
+                        next() {
+                            throw new TestError();
+                        }
+                    }
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case8);
+            }, TestError);
+
+            var case9 = {
+                [Symbol.iterator]() {
+                    return {
+                        next() {
+                            return undefined;
+                        }
+                    }
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case9);
+            }, TypeError);
+
+            var case10 = {
+                [Symbol.iterator]() {
+                    return {
+                        next() {
+                            return "a string";
+                        }
+                    }
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case10);
+            }, TypeError);
+
+            var case11 = {
+                [Symbol.iterator]() {
+                    return {
+                        next() {
+                            return {
+                                get done() {
+                                    throw new TestError();
+                                }
+                            };
+                        }
+                    }
+                }
+            };
+
+            assert.throws(() => {
+                var obj = new AggregateError(case8);
+            }, TestError);
+

```

Comment:
Need cases for non-objects like AggregateError(0), AggregateError("string") etc.